util/compression/compression.go

```diff
@@ -27,10 +27,18 @@ const (
 	// EStargz is used for estargz data.
 	EStargz
 
+	// Zstd is used for Zstandard data.
+	Zstd
+
 	// UnknownCompression means not supported yet.
 	UnknownCompression Type = -1
 )
 
+const (
+	mediaTypeDockerSchema2LayerZstd = images.MediaTypeDockerSchema2Layer + ".zstd"
```

Comment:
Do we need to support `application/vnd.docker.image.rootfs.diff.tar.zstd`? Maybe we can forcibly turn on `oci-mediatype=true` when zstd.

Reply:
IIUC no `application/vnd.docker.image.rootfs.diff.tar.zstd` in https://github.com/distribution/distribution/blob/v2.7.1/docs/spec/manifest-v2-2.md#media-types ?

Reply:
I think this can provide more compatibility with registries. Not hard to support it and impossible for the spec to define something incompatible.

Reply:
I'd prefer to keep this OCI-only.

I guess non-OCI compliant registries are unlikely to support `MediaTypeDockerSchema2Layer + ".zstd"` too?

Reply:
Docker hub seems to support the docker mediatypes without any changes. I would guess at least Moby should always support them as well (after general zstd update).

Reply:
fwiw, zstd is not part of any standard atm. Even OCI has never released it. I guess if `distribution/distribution` would ever add this mediatype (I'm not sure if they are interested in maintaining this part) it wouldn't be before docker engine actually supports it.